File: project/flutter_application_1/backend/models/Transaction.js
const mongoose = require('mongoose');

const TransactionSchema = new mongoose.Schema({
  description: { type: String, required: true },
  amount: { type: Number, required: true },
  date: { type: Date, default: Date.now },
  debitAccountId: { type: mongoose.Schema.Types.ObjectId, ref: 'Account', required: true },
  creditAccountId: { type: mongoose.Schema.Types.ObjectId, ref: 'Account', required: true },
  type: { type: String, enum: ['Journal', 'Invoice', 'Payment', 'Expense'], default: 'Journal' },
});

module.exports = mongoose.model('Transaction', TransactionSchema);

File: project/flutter_application_1/lib/features/accounting/domain/entities/transaction.dart
import 'package:equatable/equatable.dart';

class Transaction extends Equatable {
  final String id;
  final String description;
  final double amount;
  final DateTime date;
  final String debitAccountId;
  final String creditAccountId;
  final String type; // Journal, Invoice, Payment, etc.

  const Transaction({
    required this.id,
    required this.description,
    required this.amount,
    required this.date,
    required this.debitAccountId,
    required this.creditAccountId,
    required this.type,
  });

  @override
  List<Object?> get props => [
        id,
        description,
        amount,
        date,
        debitAccountId,
        creditAccountId,
        type,
      ];
}
